### PR TITLE
Introduce DependenciesFor

### PR DESCRIPTION
**Pull request by mrvoorhe**
So that collection of dependencies can be customized by other linkers

### CODE REVIEW

File: linker/Mono.Linker/LinkContext.cs [hunk 1]
```diff
@@ -194,6 +195,19 @@ public void SafeReadSymbols (AssemblyDefinition assembly)
 			} catch {}
 		}
 
+		public virtual ICollection<AssemblyDefinition> DependenciesFor (AssemblyDefinition assembly)
```
Comment (marek-safar) [inline linker/Mono.Linker/LinkContext.cs]: Use name ResolveReferences instead

File: linker/Mono.Linker/LinkContext.cs [hunk 2]
```diff
@@ -28,8 +28,9 @@
 
 using System;
 using System.Collections;
+using System.Collections.Generic;
 using System.IO;
-
+using System.Linq;
```
Comment (marek-safar) [inline linker/Mono.Linker/LinkContext.cs]: Why do you need this using?

Comment (mrvoorhe) [inline linker/Mono.Linker/LinkContext.cs]: Removed

**Comment by mrvoorhe** [comment]
@marek-safar Using removed and method renamed

**Pull request merged**